django/utils/timesince.py

```diff
@@ -14,14 +13,16 @@
     "minute": ngettext_lazy("%(num)d minute", "%(num)d minutes", "num"),
 }
 
-TIMESINCE_CHUNKS = (
-    (60 * 60 * 24 * 365, "year"),
-    (60 * 60 * 24 * 30, "month"),
-    (60 * 60 * 24 * 7, "week"),
-    (60 * 60 * 24, "day"),
-    (60 * 60, "hour"),
-    (60, "minute"),
-)
+TIME_STRING_KEYS = list(TIME_STRINGS.keys())
```

Comment:
Does this still work correctly if I pass a custom `time_strings` to `timesince`? Could you show that with a test please?

Reply:
Thanks for this, I missed that point. At first I though "oh damn! I forgot about the custom `time_strings`!" And made the keys local as you suggested.

But that was also incorrect, as it would cause weird behaviors. For example, if the custom `time_strings` has the keys in a different orders.  If the given dictionary is ordered in reverse, and there is one year between the two dates, the output would actually be "one minute". 

I think that, even if it's possible to receive custom `time_strings`, it doesn't make sense for it to not have at least that set of keys. And the reason why tests were passing is that, as long as the custom `time_strings` contains the required keys, the algorithm works fine.

So yes, I can add some more tests, but what is the expected behavior in case the function receives nonsense or incomplete dictionaries? 

Currently, if the algorithm detects that there is one year between two dates, but the "year" key is not in the `time_strings` dictionary, if fails with `KeyError: 'year'`.